### PR TITLE
Implement all API functions in PL/pgSQL

### PR DESCRIPTION
**Pull request by v0idpwn**
Progress for #199

### CODE REVIEW

**Comment by v0idpwn** [comment]
@ChuckHend I'm getting `PRIMARY KEY constraint on table \"q_test_duration_804\" lacks column \"enqueued_at\" which is part of the partition key.` for the case where the partition column is `enqueued_at`. It seems to me that I've the exact same queries but there must be some oversight. Do you have any idea of what could be causing this?

**Comment by v0idpwn** [comment]
Fixed the `is_partitioned` issue. 

I'm able to reproduce it locally, it only happens when you use a time-based partition key, for example:

```
pgmq=# select * from pgmq.create_partitioned('asd9', 'daily', 'weekly');
ERROR:  unique constraint on partitioned table must include all partitioning columns
DETAIL:  PRIMARY KEY constraint on table "q_asd9" lacks column "enqueued_at" which is part of the partition key.
```

**Comment by ChuckHend** [comment]
Strange, I can reproduce it now too. Also, executing the `CREATE TABLE` statement that is shown in the error directly also produces the same error.

```
pgmq=# select pgmq.create_partitioned('part_2', 'daily', 'weekly');
ERROR:  unique constraint on partitioned table must include all partitioning columns
DETAIL:  PRIMARY KEY constraint on table "q_part_2" lacks column "enqueued_at" which is part of the partition key.
CONTEXT:  SQL statement "
        CREATE TABLE IF NOT EXISTS pgmq.q_part_2 (
            msg_id BIGINT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
            read_ct INT DEFAULT 0 NOT NULL,
            enqueued_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
            vt TIMESTAMP WITH TIME ZONE NOT NULL,
            message JSONB
        ) PARTITION BY RANGE (enqueued_at)
    "
PL/pgSQL function pgmq.create_partitioned(text,text,text) line 9 at EXECUTE
```

**Comment by ChuckHend** [comment]
I wonder if this is has anything to do with creating the partitioned index within the same transaction that the table is created?

**Comment by ChuckHend** [comment]
@v0idpwn - the current create statement for partition queue does not make `msg_id` a primary key during `create table`

https://github.com/tembo-io/pgmq/blob/b214772c22f69411fd672ba9e13fd256f56142ee/src/partition.rs#L82

Removing the pkey from the create statement seems to let make this work. I wonder if this is a bug on `main` branch though, and we should be running something like this when partitioning by a time range.

```sql
CREATE TABLE IF NOT EXISTS pgmq.q_myqueue (
    msg_id BIGINT GENERATED ALWAYS AS IDENTITY,
    read_ct INT DEFAULT 0 NOT NULL,
    enqueued_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
    vt TIMESTAMP WITH TIME ZONE NOT NULL,
    message JSONB,
    PRIMARY KEY (enqueued_at, msg_id) <-- new
) PARTITION BY RANGE (enqueued_at);
```

**Comment by v0idpwn** [comment]
This indeed works! I didn't see the difference. Thanks!

**Comment by v0idpwn** [comment]
I think this is complete.
There's further work to be done but it could be done in future PRs.

**Review by vrmiguel — APPROVED** [review]

**Review by ChuckHend — APPROVED** [review]
Great work @v0idpwn

File: src/sql_src.sql [hunk 1]
```diff
@@ -337,3 +344,433 @@ BEGIN
...
+        num := partition_interval::INTEGER;
+        RETURN 'msg_id';
+    EXCEPTION
+        WHEN others THEN
+            RETURN 'enqueued_at';
+    END;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION pgmq.ensure_pg_partman_installed()
```
Comment (ChuckHend) [inline src/sql_src.sql:659]: should this maybe be `_ensure_pg_partman_installed()` similar to `_get_partitioned_col()`?

Comment (v0idpwn) [reply src/sql_src.sql:659]: I think it's a good idea.

Comment (v0idpwn) [reply src/sql_src.sql]: Done in https://github.com/tembo-io/pgmq/pull/212/commits/70121647174367d35e409b72d17108701be0b1dc

**Pull request merged**